### ISSUE TITLE
nan loss values during training

### ISSUE THREAD

**Issue by FicekD** [zzh8829/yolov3-tf2#233]
I checked the other issues regarding this topic but I couldn't solve my problem. I am unable to train, all loss indicators during training are `nan`.

I'm training on a custom dataset with 4 classes with slightly modified `train.py`. Weights are converted according to the tutorial, I also tried to run the tutorial on VOC2012 and it worked fine.

Python call:
```
python train.py \
	--dataset ./data/tf_record_train.tfrecord \
	--val_dataset ./data/tf_record_val.tfrecord \
	--classes ./data/marks.names \
	--num_classes 4 \
	--mode fit --transfer darknet \
	--batch_size 5 \
	--epochs 5 \
	--weights ./checkpoints/yolov3.tf \
	--weights_num_classes 80 
```
Everything else is default, did not modify default values.

Modified lines in `train.py` (I hope i didn't miss any):
```python
# line 58
model = YoloV3(FLAGS.size, training=True, classes=4)
```
```python
# line 95
model_pretrained = YoloV3(FLAGS.size, training=True, classes=80)
```
```python
# line 123
loss = [YoloLoss(anchors[mask], classes=4) for mask in anchor_masks]
```

With these changes `num_classes` and `weights_num_classes` arguments should lose their meaning.

marks.names file:
```
BottomLeft
BottomRight
TopLeft
TopRight
```

Example of `tfrecord`:
```
features {
  feature {
    key: "image/encoded"
    value {
      bytes_list {
        value: "\211PNG\r\n\032\n\ <more_values>"
      }
    }
  }
  feature {
    key: "image/filename"
    value {
      bytes_list {
        value: "BL_0_TR_62_BR_60_TL_2_.png"
      }
    }
  }
  feature {
    key: "image/format"
    value {
      bytes_list {
        value: "png"
      }
    }
  }
  feature {
    key: "image/height"
    value {
      int64_list {
        value: 416
      }
    }
  }
  feature {
    key: "image/key/sha256"
    value {
      bytes_list {
        value: "dc834637dd97f6973f19a2855d05ac189211ab056c321829e20bc9704498997f"
      }
    }
  }
  feature {
    key: "image/object/bbox/xmax"
    value {
      float_list {
        value: 0.3461538553237915
        value: 0.8293269276618958
      }
    }
  }
  feature {
    key: "image/object/bbox/xmin"
    value {
      float_list {
        value: 0.19951923191547394
        value: 0.5504807829856873
      }
    }
  }
  feature {
    key: "image/object/bbox/ymax"
    value {
      float_list {
        value: 0.4110576808452606
        value: 0.41826921701431274
      }
    }
  }
  feature {
    key: "image/object/bbox/ymin"
    value {
      float_list {
        value: 0.26442307233810425
        value: 0.19230769574642181
      }
    }
  }
  feature {
    key: "image/object/class/label"
    value {
      int64_list {
        value: 0
        value: 3
      }
    }
  }
  feature {
    key: "image/object/class/text"
    value {
      bytes_list {
        value: "BottomLeft"
        value: "TopRIght"
      }
    }
  }
  feature {
    key: "image/source_id"
    value {
      bytes_list {
        value: "BL_0_TR_62_BR_60_TL_2_.png"
      }
    }
  }
  feature {
    key: "image/width"
    value {
      int64_list {
        value: 416
      }
    }
  }
}
features {
<another_image_annotation>
}
<more_images_annotations>
```

Callbacks during training:

- After first epoch
```
[[{{node IteratorGetNext}}]]
2020-04-15 21:51:54.725860: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Out of range: End of sequence
	 [[{{node IteratorGetNext}}]]
	 [[IteratorGetNext/_8]]
2020-04-15 21:52:00.118671: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Out of range: End of sequence
	 [[{{node IteratorGetNext}}]]
2020-04-15 21:52:00.119143: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Out of range: End of sequence
	 [[{{node IteratorGetNext}}]]
	 [[IteratorGetNext/_2]]
C:\Users\dfice\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\keras\callbacks.py:1916: RuntimeWarning: invalid value encountered in less
  self.monitor_op = lambda a, b: np.less(a, b - self.min_delta)

Epoch 00001: saving model to checkpoints/yolov3_train_1.tf
C:\Users\dfice\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow_core\python\keras\callbacks.py:1261: RuntimeWarning: invalid value encountered in less
  if self.monitor_op(current - self.min_delta, self.best):
```
- After second epoch
```
2020-04-15 21:52:18.622583: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Out of range: End of sequence
	 [[{{node IteratorGetNext}}]]
2020-04-15 21:52:18.622966: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Out of range: End of sequence
	 [[{{node IteratorGetNext}}]]
	 [[IteratorGetNext/_2]]

Epoch 00002: saving model to checkpoints/yolov3_train_2.tf

23/23 [==============================] - 18s 779ms/step - loss: nan - yolo_output_0_loss: nan - yolo_output_1_loss: nan - yolo_output_2_loss: nan - val_loss: nan - val_yolo_output_0_loss: nan - val_yolo_output_1_loss: nan - val_yolo_output_2_loss: nan
WARNING:tensorflow:Unresolved object in checkpoint: (root).layer-8
W0415 21:52:22.357214  9088 util.py:144] Unresolved object in checkpoint: (root).layer-8
WARNING:tensorflow:Unresolved object in checkpoint: (root).layer-9
W0415 21:52:22.357214  9088 util.py:144] Unresolved object in checkpoint: (root).layer-9
WARNING:tensorflow:Unresolved object in checkpoint: (root).layer-10
W0415 21:52:22.357214  9088 util.py:144] Unresolved object in checkpoint: (root).layer-10
WARNING:tensorflow:Unresolved object in checkpoint: (root).layer-11
W0415 21:52:22.357214  9088 util.py:144] Unresolved object in checkpoint: (root).layer-11
WARNING:tensorflow:A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details.
W0415 21:52:22.357214  9088 util.py:152] A checkpoint was restored (e.g. tf.train.Checkpoint.restore or tf.keras.Model.load_weights) but not all checkpointed values were used. See above for specific issues. Use expect_partial() on the load status object, e.g. tf.train.Checkpoint.restore(...).expect_partial(), to silence these warnings, or use assert_consumed() to make the check explicit. See https://www.tensorflow.org/guide/checkpoint#loading_mechanics for details.

```

**Comment by FicekD** [comment]
Ok looks like I have a typo in my tfrecord file where `"TopRIght"` is supposed to be `"TopRight"`. This solved my issue.
